src/_guides/language/effective-dart/usage.md

```diff
@@ -1156,6 +1233,54 @@ Future<bool> fileContainsBear(String path) async {
 }
 {% endprettify %}
 
+
+### DO test for a future when disambiguating a `FutureOr<T>` whose value type could be `Object`.
```

Comment:
That works, but it might be overly specific. If you have a `FutureOr<int>`, it's perfectly fine to test for `is int` and might give you cleaner code do to that.